system/git_config.py

```diff
@@ -0,0 +1,216 @@
+- git_config: name=color.ui value=auto
+
+# Make etckeeper not complain when invoked by cron
+- git_config: name=user.email repo=/etc scope=local value="root@{{ ansible_fqdn }}"
+
+# Read individual values from git config
+- git_config: name=alias.ci scope=global
+# scope=system is also assumed when reading values, unless list_all=yes
+- git_config: name=alias.diffc
+
+# Read all values from git config
+- git_config: list_all=yes scope=global
+# When list_all=yes and no scope is specified, you get configuration from all scopes
+- git_config: list_all=yes
+# Specify a repository to include local settings
+- git_config: list_all=yes repo=/path/to/repo.git
+'''
+
+RETURN = '''
+---
+config_value:
+  description: When list_all=no and value is not set, a string containing the value of the setting in name
+  returned: success
+  type: string
+  sample: "vim"
+
+config_values:
+  description: When list_all=yes, a dict containing key/value pairs of multiple configuration settings
+  returned: success
+  type: dictionary
+  sample:
+    core.editor: "vim"
+    color.ui: "auto"
+    alias.diffc: "diff --cached"
+    alias.remotev: "remote -v"
+'''
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            list_all=dict(required=False, type='bool', default=False),
+            name=dict(type='str'),
+            repo=dict(type='str'),
+            scope=dict(required=False, type='str', choices=['local', 'global', 'system']),
+            value=dict(required=False)
+        ),
+        mutually_exclusive=[['list_all', 'name'], ['list_all', 'value']],
+        required_if=[('scope', 'local', ['repo'])],
+        required_one_of=[['list_all', 'name']],
+        supports_check_mode=True,
+    )
+    git_path = module.get_bin_path('git')
+    if not git_path:
+        module.fail_json(msg="Could not find git. Please ensure it is installed.")
+
+    params = module.params
+
+    if params['name']:
+        name = params['name']
+    else:
+        name = None
+
+    if params['scope']:
+        scope = params['scope']
+    elif params['list_all']:
+        scope = None
+    else:
+        scope = 'system'
+
+    if params['value']:
+        new_value = params['value']
+    else:
+        new_value = None
+
+    args = [git_path, "config"]
+    if params['list_all']:
+        args.append('-l')
+    if scope:
+        args.append("--" + scope)
+    if name:
+        args.append(name)
+
+    if scope == 'local':
+        dir = params['repo']
+    elif params['list_all'] and params['repo']:
+        # Include local settings from a specific repo when listing all available settings
+        dir = params['repo']
+    else:
+        # Run from root filesystem to avoid accidentally picking up any local config settings
+        dir = "/"
+
+    (rc, out, err) = module.run_command(' '.join(args), cwd=dir)
+    if params['list_all'] and scope and rc == 128 and 'unable to read config file' in err:
```

Comment:
Since you look at the error message, you need to make sure the locales are properly set ( https://github.com/ansible/ansible-modules-core/pull/3257 , https://github.com/ansible/ansible-modules-core/pull/3286 for example ). Since ssh also transmit the locales from the client, this can impact remote systems.